test-e2e/device-info.js

```diff
@@ -19,3 +20,35 @@ test('write and read deviceInfo', async (t) => {
   const readInfo2 = await manager.getDeviceInfo()
   t.alike(readInfo2, info2)
 })
+
+test('sending and receiving device info on initial connection', async (t) => {
+  t.plan(4)
+
+  const us = new MapeoManager({
+    rootKey: KeyManager.generateRootKey(),
+    dbFolder: ':memory:',
+    coreStorage: () => new RAM(),
+  })
+
+  await us.setDeviceInfo({ name: 'us' })
+
+  const them = new MapeoManager({
+    rootKey: KeyManager.generateRootKey(),
+    dbFolder: ':memory:',
+    coreStorage: () => new RAM(),
+  })
+
+  await them.setDeviceInfo({ name: 'them' })
+
+  us[kRPC].on('device-info', (deviceInfo) => {
+    t.is(deviceInfo.deviceId, them.deviceId)
+    t.is(deviceInfo.name, 'them')
+  })
+
+  them[kRPC].on('device-info', (deviceInfo) => {
+    t.is(deviceInfo.deviceId, us.deviceId)
```

Comment:
the `deviceId` checks here fail due to the `deviceInfo.deviceId` representing the hex string of the peer's remote public key, which seems to be derived differently than our `MapeoManager.deviceId` property

Reply:
for posterity, this was related to the `replicate` rpc helper, which auto generates the keypair used to derive the device id. in practice, it would be aligned with the keypair used by the manager+projects